packages/purpose-process/src/services/validators.ts

```diff
@@ -29,3 +45,96 @@ export const isRiskAnalysisFormValid = (
 
 export const purposeIsDraft = (purpose: Purpose): boolean =>
   !purpose.versions.some((v) => v.state !== purposeVersionState.draft);
+
+export const isEserviceMode = (
+  eservice: EService,
+  expectedMode: EServiceMode
+): void => {
+  if (eservice.mode !== expectedMode) {
+    throw eServiceModeNotAllowed(eservice.id, expectedMode);
+  }
+};
+
+export const isFreeOfCharge = (
+  isFreeOfCharge: boolean,
+  freeOfChargeReason: string | undefined
+): void => {
+  if (isFreeOfCharge && !freeOfChargeReason) {
+    throw missingFreeOfChargeReason();
+  }
+};
```

Comment:
Could we make the name of function more meaningful? For example: `assertConsistentFreeOfCharge`